### DISCUSSION TITLE
TinyUF2 to the WeAct Black Pill F411 using ST-Link.

### DISCUSSION THREAD

**Discussion by jys923** [adafruit/tinyuf2#422] [Q&A]
I want to upload TinyUF2 to the WeAct Black Pill F411 using ST-Link. 
It seems like it's done with CubeProgrammer, 
but could you explain the process in detail? 
Which file format should I use: .bin, .hex, or .uf2? 
Should I upload it to address 0x00 or 0x800000, or somewhere else?

![image](https://github.com/user-attachments/assets/a2bbb397-5a49-4602-b696-dc21f9bd74b1)
![image](https://github.com/user-attachments/assets/5cc063ae-46ec-4019-915a-e8f265f7e1ec)

I uploaded the .uf2 file starting from address 0x800000. However, no matter which button I double-tap, I can't get a drive to appear in Windows Explorer where I can drag and drop the .uf2 file. The memory map is as shown in the picture. It seems like it was uploaded correctly.

**Comment by drashna** [comment]
you should be able to use dfu-util to flash it.  But for cube programmer, you want to use the bin and 0x08000000.  At least, that's what has worked for me.

Also, if you have the 8MHz HSE crystal, you'll have to change the board.mk to use the correct value.
